kirkwood: add support for Lenovo ix2-dl

jdwl1o1 opened the pull request:
Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv
```
For USB Boot:  
``` 
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
```
For TFTP boot:
```
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000
```
3. ssh to openwrt and sysupgrade to install into flash
```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    or load from tftp using tftpboot
    sysupgrade -n /tmp/sysupgrade.bin
```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

mhei committed:
bcm27xx-userland: factor out a -dev package

Installing headers and static libraries to the target system seems
to be not required for most use cases, so let's factor them
out into a dedicated -dev package.

This cuts down to disk usage to around 50% of the original
package to ~ 2MB - not that disk space is an issue normally,
but when using inside an initramfs only project, it counts.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

mhei committed:
bcm27xx-userland: factor out a -dev package

Installing headers and static libraries to the target system seems
to be not required for most use cases, so let's factor them
out into a dedicated -dev package.

This cuts down to disk usage to around 50% of the original
package to ~ 2MB - not that disk space is an issue normally,
but when using inside an initramfs only project, it counts.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
busybox: fix compilation with GCC 10

When compiling busybox with GCC 10 and CONFIG_PKG_ASLR_PIE_ALL=y, there
are hundreds of errors like:

relocation R_MIPS16_26 against `xzalloc' cannot be used when making a
shared object; recompile with -fPIC

Simply solve this by no longer disabling PKG_ASLR_PIE, so that $(FPIC)
is properly added to the CFLAGS and LDFLAGS.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

stintel committed:
busybox: fix compilation with GCC 10

When compiling busybox with GCC 10 and CONFIG_PKG_ASLR_PIE_ALL=y, there
are hundreds of errors like:

relocation R_MIPS16_26 against `xzalloc' cannot be used when making a
shared object; recompile with -fPIC

Simply solve this by no longer disabling PKG_ASLR_PIE, so that $(FPIC)
is properly added to the CFLAGS and LDFLAGS.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
toolchain/gcc: switch to version 10 by default

Runtime-tested on:
* ath79
* bcm27xx/bcm2708
* bcm27xx/bcm2709
* bcm27xx/bcm2711
* mvebu/cortexa53
* octeon
* realtek
* x86/64

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
toolchain/gcc: switch to version 10 by default

Runtime-tested on:
* ath79
* bcm27xx/bcm2708
* bcm27xx/bcm2709
* bcm27xx/bcm2711
* mvebu/cortexa53
* octeon
* realtek
* x86/64

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
toolchain/binutils: switch to version 2.36.1 by default

Runtime-tested on:
* ath79
* bcm27xx/bcm2708
* bcm27xx/bcm2709
* bcm27xx/bcm2711
* mvebu/cortexa53
* octeon
* realtek
* x86/64

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
toolchain/binutils: switch to version 2.36.1 by default

Runtime-tested on:
* ath79
* bcm27xx/bcm2708
* bcm27xx/bcm2709
* bcm27xx/bcm2711
* mvebu/cortexa53
* octeon
* realtek
* x86/64

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

yangfl committed:
ramips: add WPS button for newifi d1

This device has a WPS button under WiFi antenna cover, add it to dts.

Signed-off-by: David Yang <mmyangfl@gmail.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

yangfl committed:
ramips: add WPS button for newifi d1

This device has a WPS button under WiFi antenna cover, add it to dts.

Signed-off-by: David Yang <mmyangfl@gmail.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: mt7628: move mtd-eeprom out of root DTSI

The mt76x8 subtarget is the only one in ramips that stores the
mediatek,mtd-eeprom property directly in the "root" mt7628an.dtsi.

This is not optimal for a few different reasons:

 * If you don't really know it or are used to other (sub)targets,
   the property will be set somewhat magically.
 * The property is set based on &factory partition before (if at all)
   this partition is defined.
 * There are several devices that have different offset or even
   different partitions to read from, which will then be overwritten
   in the DTS files. Thus, definitions are scattered between root
   DTSI and individual files.

Based on these circumstances, the "root" definition is removed and
the property is added to the device-based DTS(I) files where needed
and applicable. This should be easier to grasp for unexperienced
developers and will move the property closer to the partition
definitions.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

adschm committed:
ramips: convert most mtd-mac-address cases in DTSI to nvmem

Convert most of the cases from mtd-mac-address to nvmem where
MAC addresses are set in the DTSI, but the partitions are only
located in the device DTS. This posed some problems earlier, since
in these cases we are using partitions before they are defined,
and the nvmem system did not seem to like that.

There have been a few different resolution approaches, based on
the different tradeoffs of deduplication vs. maintainability:

 1. In many cases, the partition tables were identical except for
    the firmware partition size, and the firmware partition was
    the last in the table.
    In these cases, the partition table has been moved to the
    DTSI, and only the firmware partition's "reg" property has
    been kept in the DTS files. So, the updated nvmem definition
    could stay in the DTSI files as well.

 2. For all other cases, splitting up the partition table would
    have introduced additional complexity. Thus, the nodes to be
    converted to nvmem have been moved to the DTS files where the
    partitioning was defined.

 3. For Netgear EX2700 and WN3000RP v3, the remaining DTSI file
    was completely dissolved, as it was quite small and the name
    was not really nice either.

 4. The D-Link DIR-853 A3 was converted to nvmem as well, though
    it is just a plain DTS file not taken care of in the first
    wave.

In addition, some minor rearrangements have been made for tidyness.

Not covered (yet) by this patch are:

 * Various unielec devices
 * The D-Link DIR-8xx family

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: convert most mtd-mac-address cases in DTSI to nvmem

Convert most of the cases from mtd-mac-address to nvmem where
MAC addresses are set in the DTSI, but the partitions are only
located in the device DTS. This posed some problems earlier, since
in these cases we are using partitions before they are defined,
and the nvmem system did not seem to like that.

There have been a few different resolution approaches, based on
the different tradeoffs of deduplication vs. maintainability:

 1. In many cases, the partition tables were identical except for
    the firmware partition size, and the firmware partition was
    the last in the table.
    In these cases, the partition table has been moved to the
    DTSI, and only the firmware partition's "reg" property has
    been kept in the DTS files. So, the updated nvmem definition
    could stay in the DTSI files as well.

 2. For all other cases, splitting up the partition table would
    have introduced additional complexity. Thus, the nodes to be
    converted to nvmem have been moved to the DTS files where the
    partitioning was defined.

 3. For Netgear EX2700 and WN3000RP v3, the remaining DTSI file
    was completely dissolved, as it was quite small and the name
    was not really nice either.

 4. The D-Link DIR-853 A3 was converted to nvmem as well, though
    it is just a plain DTS file not taken care of in the first
    wave.

In addition, some minor rearrangements have been made for tidyness.

Not covered (yet) by this patch are:

 * Various unielec devices
 * The D-Link DIR-8xx family

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

CHKDSK88 commented on target/linux/kirkwood/base-files/etc/board.d/02_network:
```diff
@@ -35,6 +35,9 @@ linksys,ea4500)
 	ucidef_set_interfaces_lan_wan "ethernet1 ethernet2 ethernet3 ethernet4" "internet"
 	ucidef_set_interface_macaddr "wan" $( mtd_get_mac_ascii u_env eth1addr )
 	;;
+"lenovo,ix2")
```
duplicate section

CHKDSK88 commented on target/linux/kirkwood/base-files/etc/diag.sh:
```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
```
Please drop it. Status led are configured in dts. Please see an example:
https://github.com/openwrt/openwrt/blob/ee1a92834da4dab409dc4cd150bb188c6b069020/target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-ea3500.dts#L26

jdwl1o1 replied:
Removed

CHKDSK88 commented on target/linux/kirkwood/base-files/etc/init.d/hwmon_fancontrol:
```diff
@@ -44,5 +44,18 @@ boot() {
 		# pre-defined steps: 103000, 122000, 143300, 170000 in [millicelsius]
 		echo 103000 > "$path_to_hwmon/temp3_auto_temp_max"
 		;;
+	lenovo,ix2)
```
alphabetical order missed

jdwl1o1 replied:
Set to Alphabetical Order

CHKDSK88 commented on target/linux/kirkwood/patches-5.10/113-lenovo-ix2.patch:
```diff
@@ -0,0 +1,708 @@
+--- a/arch/arm/boot/dts/kirkwood-lenovo_ix2.dts
++++ b/arch/arm/boot/dts/kirkwood-lenovo_ix2.dts
```
Please change patch to regular file and move it to: target/linux/kirkwood/files/arch/arm/boot/dts/

jdwl1o1 replied:
Done, new DTS created

CHKDSK88 commented on target/linux/kirkwood/patches-5.10/113-lenovo-ix2.patch:
```diff
@@ -0,0 +1,708 @@
++/ {
++	#address-cells = < 0x01 >;
++	#size-cells = < 0x01 >;
++	compatible = "lenovo,ix2\0marvell,kirkwood-88f6282\0marvell,kirkwood";
```
Something goes wrong here.

jdwl1o1 replied:
Patch removed

CHKDSK88 commented on target/linux/kirkwood/patches-5.10/113-lenovo-ix2.patch:
```diff
@@ -0,0 +1,708 @@
++++ b/arch/arm/boot/dts/kirkwood-lenovo_ix2.dts
+@@ -0,0 +1,705 @@
++/dts-v1/;
++
```
Please rewrite dts, and use:
```
#include "kirkwood.dtsi"
#include "kirkwood-6281.dtsi"
```

jdwl1o1 replied:
Done, new DTS created and tested.

CHKDSK88 reviewed (changes requested):
Dts need a huge refactoring. It looks like decompiled from dtb. Some chamges in other file are required too.

jdwl1o1 commented:
Hi,

Thank you for the feedback. 

Unfortunately re-writing the DTS is beyond my capability and would require many more tests to confirm functionality.

stintel committed:
ipq40xx: fix Edgecore ECW5211 boot

The bootloader will look for a configuration section named ap.dk01.1-c2
in the FIT image. If this doesn't exist, the device won't boot.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
ipq40xx: fix Edgecore ECW5211 boot

The bootloader will look for a configuration section named ap.dk01.1-c2
in the FIT image. If this doesn't exist, the device won't boot.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

stintel committed:
umbim: bump to git HEAD

 de56231 umbim: fix compilation with GCC 10

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

stintel committed:
umbim: bump to git HEAD

 de56231 umbim: fix compilation with GCC 10

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

adschm committed:
ath79: convert remaining mtd-mac-address cases to nvmem

Since the nvmem-based approach for retrieving MAC addresses
appears to depend on the addresses being set up after the
partitions, it is no longer possible to keep the MAC address
setup in shared DTSI files while the partitions itself are
set up in DTS files for the individual devices.

In ath79 the firmware partition is typically located somewhere
"in the middle" of the partition table. Thus, it's not trivial
to share the partitions containing MAC address information in
a common DTSI (like we did in some cases on ramips).

In this commit, MAC address setup is thus moved to the relevant
partitions, and in most cases needs to be duplicated. While
the duplication is not really nice, it eventually provides a
cleaner and more tidy setup, making the DTS(I) file
fragmentation a bit more logical. This should also help
with adding new devices, as information is distributed across
less locations.

For consistency, this commit also moves the mtd-cal-data property
"down" together with the MAC address setup, so it's not based
on a partition before the latter is defined either. (This is
only done for those files touched due to nvmem conversion.)

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ath79: convert remaining mtd-mac-address cases to nvmem

Since the nvmem-based approach for retrieving MAC addresses
appears to depend on the addresses being set up after the
partitions, it is no longer possible to keep the MAC address
setup in shared DTSI files while the partitions itself are
set up in DTS files for the individual devices.

In ath79 the firmware partition is typically located somewhere
"in the middle" of the partition table. Thus, it's not trivial
to share the partitions containing MAC address information in
a common DTSI (like we did in some cases on ramips).

In this commit, MAC address setup is thus moved to the relevant
partitions, and in most cases needs to be duplicated. While
the duplication is not really nice, it eventually provides a
cleaner and more tidy setup, making the DTS(I) file
fragmentation a bit more logical. This should also help
with adding new devices, as information is distributed across
less locations.

For consistency, this commit also moves the mtd-cal-data property
"down" together with the MAC address setup, so it's not based
on a partition before the latter is defined either. (This is
only done for those files touched due to nvmem conversion.)

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm commented:
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes#do_not_multi-post_or_re-post_prs

jdwl1o1 committed:
IX2-DL DTS, OpenWRT formatting, refactored. V1

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
IX2-DL DTS, OpenWRT formatting, refactored. V1

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>

jdwl1o1 committed:
Add Lenovo IX2-DL, reorder alphabetically

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Add Lenovo IX2-DL, reorder alphabetically

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: convert mtd-mac-address to nvmem for D-Link DIR-8xx

Convert this series by moving the definitions to the individual
devices.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: fix indent in DTS for GL.iNet VIXMINI/microuter-N300

Use correct indent.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: remove redundant nvmem definitions

Due to use of a script when migrating from mtd-mac-address, a few
of the definitions are redundant in DTSI and DTS files. Remove
those.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

dangowrt committed:
image: fix build with SELinux

The option '-xattr' for mksquashfs4 should be '-xattrs' which lead to
build failure with SELinux enabled. Add the missing 's'.

Fixes: 4baf47b9a8 ("images: squashfs: xattrs should not depend on buld host")
Signed-off-by: Daniel Golle <daniel@makrotopia.org>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

dangowrt committed:
image: fix build with SELinux

The option '-xattr' for mksquashfs4 should be '-xattrs' which lead to
build failure with SELinux enabled. Add the missing 's'.

Fixes: 4baf47b9a8 ("images: squashfs: xattrs should not depend on buld host")
Signed-off-by: Daniel Golle <daniel@makrotopia.org>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Merge branch 'openwrt:master' into master

jdwl1o1 commented:
Hi OpenWRT team, I've updated the PR with the corrected files and new DTS. 

I am not very familiar with git - other commits have been picked up that I have not been involved with?

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
kirkwood: add support for Lenovo ix2-dl

Lenovo ix2-dl is a dual SATA NAS powered by a Marvell
 Kirkwood SoC clocked at 1.6GHz. It has 256MB of RAM and 1GiB of
 flash memory, 1x USB 2.0 and 1x 1Gbit/s NIC.
 Was also sold by Iomega as the Iomega StorCentre ix2-dl.
 This device is very similar to the ix2-200 however it lacks 2 USB ports and includes
 a larger nand flash unit. The IX2-DL was sold without drives, with the OS stored in nand.

Specification:
- SoC: Marvell Kirkwood 88F6281
- CPU/Speed: 1600Mhz
- Flash-Chip: Samsung NAND
- Flash size: nand: 1024 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
- RAM: 256MB
- LAN: 1x 1000 Mbps Ethernet
- WiFi: none
- 1x USB 2.0
- UART1: for serial console. UART0 not available.

Installation instructions (modified version of the ix2-200 commit):
This runs OpenWrt from ram to allow sysupgrade to install to nand.

1. download initramfs-uImage image and sysupgrade image, copy into tftp server or ext2 formatted usb drive
2. access uboot environment with serial cable and run
    ```
    setenv mainlineLinux yes
    setenv arcNumber 1682
    setenv console 'console=ttyS0,115200'
    setenv mtdparts 'mtdparts=orion_nand:0x100000@0x000000(u-boot)ro,0x20000@0xA0000(u-boot environment)ro,0x300000@0x100000(kernel),0x1C00000@0x400000(ubi)'
    setenv
    setenv bootargs_root 'root='
    setenv bootcmd 'setenv bootargs ${console} ${mtdparts} ${bootargs_root}; nand read.e 0x800000 0x100000 0x300000; bootm 0x00800000'
    saveenv

For USB Boot:   
 usb reset; ext2load usb 0:1 0x00800000 /initramfs.bin; bootm 0x00800000
    ```
For TFTP boot:
    setenv serverip [tftp server ip]    
    setenv ipaddr 192.168.1.13
    tftpboot 0x00800000 factory.bin
    bootm 0x00800000

3. ssh to openwrt and sysupgrade to install into flash
    ```
    mkdir /mnt/usb
    mount /dev/sda1 /mnt/usb
    cp /mnt/usb/*sysupgrade* /tmp
    sysupgrade -n /tmp/sysupgrade.bin
    ```
4. access openwrt by dhcp ip address assigned by your router or at 192.168.1.1

note 1 - sata drives should not be installed when installing
note 2 - this is a 1 way process. Once the nand is overwritten returning to stock will not be possible

Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 commented:
Hi, I have reworked the dts completely, using details from other IX2-DL user' dts files. I've also identified that there are multiple models of 'IX2' from Iomega and Lenovo and so have made this PR specific to the IX2-DL only. 
The 'non-nand' model would be similar and the dts is similar, however I do not have access to one of those to test.

adschm committed:
ath79: remove redundant nvmem definitions

Due to use of a script when migrating from mtd-mac-address, a few
of the definitions are redundant in DTSI and DTS files. Remove
those and consolidate the definitions in parent DTSI files in a
few cases.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
ramips: convert mtd-mac-address to nvmem for UniElec devices

Convert this series by moving the definitions to the individual
devices.

Now all devices on ramips are converted.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

dangowrt committed:
selinux-policy: update to version 0.9

 592ac0f add a note
 4bacd14 sslcertfile: list /etc/ssl
 7bdefa4 example: indicate that skip is an option
 d1e9a85 wifi: sys pipe usage
 eb903e1 README: add note about policycoreutils-setfiles weak dependency
 762e011 ttyd: signull all subjects
 fbfc079 acme: add basic support for acme_cleanup.sh and acme_setup.sh
 9ac7592 acme: transition to sys.subj on generic initscript execution
 f3dd1ba acme: missing rules related to sys.subj trans on file.initscriptfile
 ae273fa odhcp6c/netifd: support drop-in directories
 5fa9b41 subj: do not encourage misconfiguration
 44722b6 blockd, logd, odhcpc6, ubiutil, mtdstordev
 a775d93 21.02 related
 a473691 rcboot runs rcuhttpd which creates /tmp/etc for /tmp/etc/uhttpd
 290e9fb rcuhttpd: related to rcboot and uci-defaults
 3fc0d8b rcuhttpd: lists /etc/uci-defaults
 1f5ef48 removes ubvol.lock policy and adds move mtd/ubi partitions

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
lantiq: convert mtd-mac-address to nvmem for Fritz!Box 736x

Convert this series by moving the definitions to the individual
devices.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm committed:
mediatek: convert mtd-mac-address to nvmem for MT7629 RFB

This conversion appears to have been overlooked since it's in a
kernel patch.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

adschm commented:
What you need is "git rebase" and "git rebase -i", so you put your changes on top of current master, and then squash all "your" changes into a single commit. "git merge" almost never helps for anything (good).

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-dl.dts:
```diff
@@ -0,0 +1,48 @@
+/ {
+	model = "Lenovo IX2-dl";
+	compatible = "lenovo-ix2-dl", "marvell,kirkwood-88f6282", "marvell,kirkwood";
+
```
remove empty line here.

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-common.dtsi:
```diff
@@ -0,0 +1,221 @@
+ * warranty of any kind, whether express or implied.
+ */
+
+/dts-v1/;
```
dts-v1 is only needed once. Please look where others put it (kirdwood.dtsi, other dtsi, or dts).

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-common.dtsi:
```diff
@@ -0,0 +1,221 @@
+		pinctrl-names = "default";
+
+		white-sys {
+			label = "ix2:white:sys";
```
please remove model prefix, just "white:sys", "red:sys", etc.

Why "sys" and not "system"?

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-common.dtsi:
```diff
@@ -0,0 +1,221 @@
+};
+
+&mdio {
+	status = "okay";
```
Please add empty line after status (everywhere).

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-dl.dts:
```diff
@@ -0,0 +1,48 @@
+		label = "uboot";
+		reg = <0x0000000 0x0080000>;
+		read-only;
+	};
```
Please add empty lines between nodes.

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-dl.dts:
```diff
@@ -0,0 +1,48 @@
+	};
+	partition@400000 {
+		label = "ubi";
+		reg = <0x0400000 0x020000000>;
```
512 MiB != 1 GiB

adschm commented on target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-lenovo-ix2-dl.dts:
```diff
@@ -0,0 +1,48 @@
+
+	partition@0 {
+		label = "uboot";
+		reg = <0x0000000 0x0080000>;
```
Please use the same number of digits everywhere or remove all leading zeros.

First offset should be just 0x0

adschm commented on target/linux/kirkwood/image/Makefile:
```diff
@@ -100,6 +100,22 @@ define Device/iom_ix2-200
 endef
 TARGET_DEVICES += iom_ix2-200
 
+define Device/lenovo-ix2-dl
```
levono_ix2-dl

adschm commented on target/linux/kirkwood/image/Makefile:
```diff
@@ -100,6 +100,22 @@ define Device/iom_ix2-200
 TARGET_DEVICES += iom_ix2-200
 
+define Device/lenovo-ix2-dl
+  DEVICE_TITLE := Lenovo IX2-dl
```
DEVICE_TITLE has been replaced by DEVICE_VENDOR/DEVICE_MODEL more than a year ago.

adschm commented on target/linux/kirkwood/image/Makefile:
```diff
@@ -100,6 +100,22 @@ define Device/iom_ix2-200
+  KERNEL_IN_UBI :=
+  UBINIZE_OPTS := -E 5
+  IMAGE_SIZE := 32505856
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | check-size $$$$(IMAGE_SIZE)
```
74 chars. per line max. (everywhere).
check-size without argument

adschm commented on target/linux/kirkwood/image/Makefile:
```diff
@@ -100,6 +100,22 @@ define Device/iom_ix2-200
+  DEVICE_TITLE := Lenovo IX2-dl
+  DEVICE_DTS := kirkwood-lenovo-ix2-dl
+  DEVICE_PACKAGES += kmod-gpio-button-hotplug kmod-i2c-mv64xxx kmod-hwmon-adt7475
+  DEVICE_TYPE:=nas
```
DEVICE_TYPE cannot be set per device. Drop it.

adschm commented on target/linux/kirkwood/image/Makefile:
```diff
@@ -100,6 +100,22 @@ define Device/iom_ix2-200
+  KERNEL_SIZE := 3072k
+  KERNEL_IN_UBI :=
+  UBINIZE_OPTS := -E 5
+  IMAGE_SIZE := 32505856
```
Where does this value come from? Why not use "k"?

adschm commented on target/linux/kirkwood/base-files/etc/board.d/02_network:
```diff
@@ -35,6 +30,14 @@ linksys,ea4500)
+seagate,goflexnet|\
+zyxel,nsa310s)
+	ucidef_set_interface_lan "eth0" "dhcp"
+	;;
```
I do not understand this change.

adschm commented on target/linux/kirkwood/base-files/etc/board.d/01_leds:
```diff
@@ -12,6 +12,9 @@ case "$board" in
 	;;
+lenovo-ix2-dl)
+	ucidef_set_led_timer "power" "power" "status:white:rebuild_led" "200" "800"
+        ;;
```
broken indent.

Using ucidef_set_led_timer is discouraged for new devices. Please use led-* aliases instead like all newer devices.

adschm reviewed (changes requested):
This seems to be based on a very old sample. Please have a look at devices in the current master to improve your submission.

stintel committed:
wolfssl: fix build with GCC 10 on 32 x86 targets

Backport upstream patch to fix build with GCC 10 on 32 x86 targets.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Merge branch 'openwrt:master' into master

jdwl1o1 committed:
Corrections as requested

jdwl1o1 committed:
Corrections as requested

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "image: fix build with SELinux"

This reverts commit 88077e453d160dc7a1ca61e03f87cf00333c9046.

jdwl1o1 committed:
Revert "image: fix build with SELinux"

This reverts commit 88077e453d160dc7a1ca61e03f87cf00333c9046.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ath79: convert remaining mtd-mac-address cases to nvmem"

This reverts commit e947e261fc5e33180ae89104c0852cdadc34f5bf.

jdwl1o1 committed:
Revert "ath79: convert remaining mtd-mac-address cases to nvmem"

This reverts commit e947e261fc5e33180ae89104c0852cdadc34f5bf.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ramips: convert most mtd-mac-address cases in DTSI to nvmem"

This reverts commit 5102c1d71ca7aea6b5addecc7740d479fd621a88.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ramips: convert most mtd-mac-address cases in DTSI to nvmem"

This reverts commit 5102c1d71ca7aea6b5addecc7740d479fd621a88.

jdwl1o1 committed:
Revert "bcm27xx-userland: factor out a -dev package"

This reverts commit 2bdd1db25df31846fa00af66f471729a4d025196.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "bcm27xx-userland: factor out a -dev package"

This reverts commit 2bdd1db25df31846fa00af66f471729a4d025196.

jdwl1o1 committed:
Revert "toolchain/gcc: switch to version 10 by default"

This reverts commit d13db09ea790439cd6dc81836d6137fe7e8befc7.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "toolchain/gcc: switch to version 10 by default"

This reverts commit d13db09ea790439cd6dc81836d6137fe7e8befc7.

jdwl1o1 committed:
Revert "toolchain/binutils: switch to version 2.36.1 by default"

This reverts commit b8b41cf9ff939376c5736074df554ec1e0386f1a.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "toolchain/binutils: switch to version 2.36.1 by default"

This reverts commit b8b41cf9ff939376c5736074df554ec1e0386f1a.

jdwl1o1 committed:
Revert "umbim: bump to git HEAD"

This reverts commit 729b6fbaf247010855555bc6ec9be00164af0aea.

jdwl1o1 committed:
Revert "umbim: bump to git HEAD"

This reverts commit 729b6fbaf247010855555bc6ec9be00164af0aea.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "busybox: fix compilation with GCC 10"

This reverts commit a6a2e070229d9735adec010811f71024de79c8dd.

jdwl1o1 committed:
Revert "busybox: fix compilation with GCC 10"

This reverts commit a6a2e070229d9735adec010811f71024de79c8dd.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ramips: add WPS button for newifi d1"

This reverts commit cc143ad1ddc83a225056232d17d448f60d690528.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ramips: add WPS button for newifi d1"

This reverts commit cc143ad1ddc83a225056232d17d448f60d690528.

jdwl1o1 committed:
Revert "ipq40xx: fix Edgecore ECW5211 boot"

This reverts commit 2d45a7273c0752f82ece2fa7c3506fdcd5bd25f6.

Signed-off-by: jdwl1o1 <89111991+jdwl1o1@users.noreply.github.com>
Signed-off-by: Jeremy Lobascher <jeremy@lobascher.com>

jdwl1o1 committed:
Revert "ipq40xx: fix Edgecore ECW5211 boot"

This reverts commit 2d45a7273c0752f82ece2fa7c3506fdcd5bd25f6.

jdwl1o1 committed:
Merge branch 'openwrt:master' into master

jdwl1o1 committed:
Merge branch 'master' of https://github.com/jdwl0o1/openwrt

jdwl1o1 commented:
Hi OpenWRT team, my tree appears to be corrupted or incorrect (or I'm doing something wrong), with every update from master then sending the commits to this PR. As I'm not able to resolve this I am going to close the PR and delete my fork.
My sincere apologies for wasting your time.
